### PR TITLE
SEPA Launcher API [Tests]

### PR DESCRIPTION
**Pull request by scannillo**
### Summary of changes

 - Unit Tests to accompany the changes made in https://github.com/braintree/braintree_android/pull/812

 ### Checklist

 - ~Added a changelog entry~

### Authors
@scannillo

### CODE REVIEW

File: SEPADirectDebit/src/test/java/com/braintreepayments/api/SEPADirectDebitClientUnitTest.java [hunk 1]
```diff
@@ -169,16 +105,17 @@ public void tokenize_onCreateMandateRequestSuccess_whenMandateAlreadyApproved_on
...
         SEPADirectDebitClient sut =
-                new SEPADirectDebitClient(activity, lifecycle, braintreeClient, sepaDirectDebitApi);
-        sut.setListener(listener);
+                new SEPADirectDebitClient(braintreeClient, sepaDirectDebitApi);
 
-        sut.tokenize(activity, sepaDirectDebitRequest);
-        verify(listener).onSEPADirectDebitSuccess(nonce);
-        verify(braintreeClient).sendAnalyticsEvent("sepa-direct-debit.tokenize.success");
+        sut.tokenize(sepaDirectDebitRequest, (sepaDirectDebitResponse, error) -> {
+            assertEquals(sepaDirectDebitResponse.getNonce(), nonce);
```
Comment (sarahkoop) [inline SEPADirectDebit/src/test/java/com/braintreepayments/api/SEPADirectDebitClientUnitTest.java:111]: Just confirming - are these callback assertions being invoked? I haven't seen this done without a countdown latch before

Comment (scannillo) [reply SEPADirectDebit/src/test/java/com/braintreepayments/api/SEPADirectDebitClientUnitTest.java]: Yes, these are! (Confirmed with breakpoints & making sure the tests failed when modified)

However I pushed a commit (a1b0434bf5b3f4c123fed315201c3e2db2e1c31b) to move these tests away from the callback syntax and to instead use ArgumentCaptor, to match with the standard way in this test file.

File: SEPADirectDebit/src/test/java/com/braintreepayments/api/SEPADirectDebitClientUnitTest.java [hunk 2]
```diff
@@ -540,30 +462,33 @@ public void onBrowserSwitchResult_whenBrowserSwitchStatusSuccess_whenDeepLinkURL
...
-        SEPADirectDebitClient sut =
-                new SEPADirectDebitClient(activity, lifecycle, braintreeClient, sepaDirectDebitApi);
-        sut.getBrowserSwitchResult(activity);
-        verify(braintreeClient).getBrowserSwitchResult(activity);
-    }
-}
\ No newline at end of file
+}
+
+// Save for SEPABrowserSwitchResult_Tests
```
Comment (sarahkoop) [inline SEPADirectDebit/src/test/java/com/braintreepayments/api/SEPADirectDebitClientUnitTest.java:481]: Should this be in separate file?

File: SEPADirectDebit/src/test/java/com/braintreepayments/api/SEPADirectDebitClientUnitTest.java (hunk 1 above)
Comment (sarahkoop) [reply SEPADirectDebit/src/test/java/com/braintreepayments/api/SEPADirectDebitClientUnitTest.java]: Oh cool we maybe should move to that pattern eventually - seems cleaner. But good to go with the current pattern for now for consistency 👍

File: SEPADirectDebit/src/test/java/com/braintreepayments/api/SEPADirectDebitClientUnitTest.java (hunk 2 above)
Comment (scannillo) [reply SEPADirectDebit/src/test/java/com/braintreepayments/api/SEPADirectDebitClientUnitTest.java]: I actually think this test is covered [here](https://github.com/braintree/braintree_android/blob/a1b0434bf5b3f4c123fed315201c3e2db2e1c31b/SEPADirectDebit/src/test/java/com/braintreepayments/api/SEPADirectDebitLauncherUnitTest.java#L70-L83). Removed this commented out section in bd911a22fb4783c7388b75b678ffe3d85af69d53

**Review by sarahkoop — APPROVED** [review]

**Pull request merged**